diff --git a/Makefile b/Makefile
index af3ed1e..0e855e9 100644
--- a/Makefile
+++ b/Makefile
@@ -209,9 +209,11 @@ deploy-community-managed:
 	kubectl apply -f deploy/common
 	kubectl -n multicluster-operators delete secret appmgr-hub-kubeconfig --ignore-not-found
 	kubectl -n multicluster-operators create secret generic appmgr-hub-kubeconfig --from-file=kubeconfig=/tmp/kubeconfig
-	sed -i 's/<managed cluster name>/$(MANAGED_CLUSTER_NAME)/g' deploy/managed/operator.yaml
-	sed -i 's/<managed cluster namespace>/$(MANAGED_CLUSTER_NAME)/g' deploy/managed/operator.yaml
-	kubectl apply -f deploy/managed
+	mkdir -p munge-manifests
+	cp deploy/managed/operator.yaml munge-manifests/operator.yaml
+	sed -i 's/<managed cluster name>/$(MANAGED_CLUSTER_NAME)/g' munge-manifests/operator.yaml
+	sed -i 's/<managed cluster namespace>/$(MANAGED_CLUSTER_NAME)/g' munge-manifests/operator.yaml
+	kubectl apply -f munge-manifests/operator.yaml
 
 
 ############################################################
diff --git a/deploy/hub/application-operator.yaml b/deploy/hub/application-operator.yaml
index babc503..0963d6c 100644
--- a/deploy/hub/application-operator.yaml
+++ b/deploy/hub/application-operator.yaml
@@ -118,6 +118,7 @@ spec:
       - command:
         - /usr/local/bin/multicluster-operators-channel
         - --alsologtostderr
+        - --debug=true
         env:
         - name: WATCH_NAMESPACE
         - name: POD_NAME
diff --git a/examples/helmrepo-hub-channel/01-channel.yaml b/examples/helmrepo-hub-channel/01-channel.yaml
index 1d4d6ac..ef59cbd 100644
--- a/examples/helmrepo-hub-channel/01-channel.yaml
+++ b/examples/helmrepo-hub-channel/01-channel.yaml
@@ -5,5 +5,5 @@ metadata:
   namespace: dev
 spec:
   type: HelmRepo
-  pathname: http://kubernetes-charts.storage.googleapis.com/
+  pathname: https://charts.helm.sh/stable/
   insecureSkipVerify: true
