diff --git a/Makefile b/Makefile
index 3a37e40..d7bec79 100644
--- a/Makefile
+++ b/Makefile
@@ -158,7 +158,10 @@ apply-spoke-cr:
 	$(SED_CMD) -e "s,quay.io/open-cluster-management/registration,$(REGISTRATION_IMAGE)," -e "s,quay.io/open-cluster-management/work,$(WORK_IMAGE)," deploy/klusterlet/config/samples/operator_open-cluster-management_klusterlets.cr.yaml | $(KUBECTL) apply -f -
 
 apply-spoke-cr-kind:
-	$(KUBECTL) --kubeconfig=$(KLUSTERLET_KIND_KUBECONFIG) apply -f deploy/klusterlet/config/samples/operator_open-cluster-management_klusterlets.cr.yaml
+	mkdir -p munge-csv
+	cp deploy/klusterlet/config/samples/operator_open-cluster-management_klusterlets.cr.yaml munge-csv/operator_open-cluster-management_klusterlets.cr.yaml
+	$(SED_CMD) -e "s,cluster1,$(KIND_CLUSTER)," -i munge-csv/operator_open-cluster-management_klusterlets.cr.yaml
+	$(KUBECTL) --kubeconfig=$(KLUSTERLET_KIND_KUBECONFIG) apply -f munge-csv/operator_open-cluster-management_klusterlets.cr.yaml
 
 clean-spoke: ensure-operator-sdk
 	$(KUBECTL) delete -f deploy/klusterlet/config/samples/operator_open-cluster-management_klusterlets.cr.yaml --ignore-not-found
